File: src/main/java/com/uitnetwork/alexa/model/AlexaIntent.java
package com.uitnetwork.alexa.model;

import java.util.HashMap;
import java.util.Map;

public enum AlexaIntent {
    MY_NAME_INTENT("MyNameIntent"),
    TIME_INTENT("TimeIntent");

    private final String intent;

    AlexaIntent(String intent) {
        this.intent = intent;
    }

    private static final Map<String, AlexaIntent> alexaIntentMap = new HashMap<>();

    static {
        for (AlexaIntent alexaIntent : AlexaIntent.values()) {
            alexaIntentMap.put(alexaIntent.intent, alexaIntent);
        }
    }

    public static final AlexaIntent lookupAlexaIntent(String intent) {
        AlexaIntent alexaIntent = alexaIntentMap.get(intent);
        return alexaIntent;
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/model/AlexaIntent.kt
package com.uitnetwork.alexa.model

import com.uitnetwork.alexa.exception.AlexaException
import java.util.*

enum class AlexaIntent private constructor(private val intent: String) {
    MY_NAME_INTENT("MyNameIntent"),
    TIME_INTENT("TimeIntent");

    companion object {
        private val alexaIntentMap = HashMap<String, AlexaIntent>()
        init {
            for (alexaIntent in AlexaIntent.values()) {
                alexaIntentMap.put(alexaIntent.intent, alexaIntent)
            }
        }

        fun lookupAlexaIntent(intent: String): AlexaIntent {
            val alexaIntent = alexaIntentMap[intent]
            return alexaIntent ?: throw AlexaException("Intent: $intent is not supported!")
        }
    }
}
